plugins/civchat2-paper/src/main/java/vg/civcraft/mc/civchat2/CivChat2Manager.java

```diff
@@ -200,6 +220,28 @@ public void broadcastMessage(Player sender, String chatMessage, String messageFo
             }
         }
 
+        // Chat filter check - block message if it contains banned words
+        if (containsBannedWord(chatMessage)) {
+            //Flag inappropriate message, mute sender for 1 hour
+            sender.sendMessage(ChatColor.RED + "Your message has been flagged for inappropriate content.");
+            banSetting.setValue(sender, System.currentTimeMillis() + TimeUnit.HOURS.toMillis(1)); // 1 hour mute
+
+            // Log the filtered message to console and mods
+            String senderName = customNames.containsKey(sender.getUniqueId()) ? customNames.get(sender.getUniqueId())
+                : sender.getDisplayName();
+
+            //Set<String> players = doSendGroupMsg(sender.getUniqueId(), senderName, group, message);
+
+            Group modsGroup = GroupManager.getGroup("Mods");
```

Comment:
Might be best to add the group name as a config value for any potential other servers